### PR TITLE
New style IndexExecutor.

### PR DESCRIPTION
**Pull request by maierlars**

### CODE REVIEW

File: arangod/Aql/IndexExecutor.cpp [hunk 1]
```diff
@@ -648,64 +669,88 @@ std::pair<ExecutionState, IndexStats> IndexExecutor::produceRows(OutputAqlItemRo
...
-      if (!advanceCursor()) {
-        _input = InputAqlItemRow{CreateInvalidInputRowHint{}};
-        // just to validate that after continue we get into retry mode
-        TRI_ASSERT(!_input);
-        continue;
+  AqlCall clientCall = output.getClientCall();
+  LOG_DEVEL_IDX << "IndexExecutor::produceRows " << clientCall;
+
+  /*
+   * Logic of this executor is as follows:
```
Comment (mchacki) [inline arangod/Aql/IndexExecutor.cpp]: yay \o/

**Comment by mchacki** [comment]
Jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8525/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8533/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8558/

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8560/

**Pull request merged**